src/components/Messages/PreviewList.tsx

```diff
@@ -27,8 +27,8 @@ const PreviewList: FC = () => {
   const showLoading = loading && (messages.size === 0 || profiles.size === 0);
 
   const sortedProfiles = Array.from(profiles).sort(([, a], [, b]) => {
-    const messageA = messages.get(a.ownedBy.toLowerCase());
-    const messageB = messages.get(b.ownedBy.toLowerCase());
+    const messageA = messages.get(profileToKey.get(a.id) ?? '');
```

Comment:
I don't think we need this new `profileToKey` map in the store. You should be able to use `buildConversationKey` to get the key with the given data and then we don't have to worry about keeping another map in sync.

```js
    const conversationId = buildConversationId(currentProfile.id, profile.id);
    const conversationKey = buildConversationKey(profile.ownedBy, conversationId);
```